src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -42,58 +55,104 @@
+
+    public static final Operator DEFAULT_LOW_FREQ_OCCUR = Operator.OR;
+
+    public static final boolean DEFAULT_DISABLE_COORD = true;
+
     public static enum Operator {
-        OR, AND
+        OR, AND;
+
+        public Occur toMustOrShouldClause() {
+            return this.equals(Operator.AND) ? Occur.MUST : Occur.SHOULD;
+        }
+
+        public static Operator readFrom(StreamInput in, Operator defaultOperator) throws IOException {
+            Integer operator = in.readOptionalVInt();
+            if (operator == null) {
+                return defaultOperator;
+            } 
+            return operator == 1 ? AND : OR;
+        }
+
+        public static void writeTo(StreamOutput out, Operator operator, Operator defaultOperator) throws IOException {
+            if (operator == null || operator == defaultOperator) {
+                out.writeBoolean(false);
+            } else {
+                out.writeOptionalVInt(operator.equals(AND) ? 1 : 0);
+            }
+        }
+
+        public static Operator fromString(String op) {
+            if ("or".equalsIgnoreCase(op)) {
+                return Operator.OR;
+            } else if ("and".equalsIgnoreCase(op)) {
+                return Operator.AND;
+            } else {
+                throw new ElasticsearchException("Operator needs to be either 'and' or 'or', but not [" + op + "]");
+            }
+        }
     }
 
-    private final String name;
+    private final String fieldName;
 
     private final Object text;
 
-    private Operator highFreqOperator = null;
+    private Operator highFreqOperator = DEFAULT_HIGH_FREQ_OCCUR;
 
-    private Operator lowFreqOperator = null;
+    private Operator lowFreqOperator = DEFAULT_LOW_FREQ_OCCUR;
 
     private String analyzer = null;
 
-    private Float boost = null;
+    private float boost = 1.0f;
 
     private String lowFreqMinimumShouldMatch = null;
 
     private String highFreqMinimumShouldMatch = null;
 
-    private Boolean disableCoords = null;
+    private Boolean disableCoord = DEFAULT_DISABLE_COORD;
 
-    private Float cutoffFrequency = null;
+    private Float cutoffFrequency = DEFAULT_CUTOFF_FREQ;
 
     private String queryName;
 
     static final CommonTermsQueryBuilder PROTOTYPE = new CommonTermsQueryBuilder();
 
     /**
+     * private constructor used only internally
+     */
+    private CommonTermsQueryBuilder() {
+        this.text = null;
+        this.fieldName = null;
+    }
+
+    /**
      * Constructs a new common terms query.
      */
-    public CommonTermsQueryBuilder(String name, Object text) {
-        if (name == null) {
+    public CommonTermsQueryBuilder(String fieldName, Object text) {
+        if (fieldName == null) {
             throw new IllegalArgumentException("Field name must not be null");
         }
         if (text == null) {
             throw new IllegalArgumentException("Query must not be null");
         }
+        this.fieldName = fieldName;
         this.text = text;
```

Comment:
I wonder if this class could extend from BaseTermQueryBuilder. There we also made sure that if a string is passed in it is converted to BytesRef at set time, otherwise strings could be either bytesref if they come from the wire/json or strings if they get set through java api, which is confusing.